### ISSUE TITLE
Getting NullPointerException when calling Distribute.goToUnknownAppsSettings

### ISSUE THREAD

**Issue by fabiendem** [microsoft/appcenter-sdk-android#1477]
### **Description**

We are getting few crash reports with the following stack trace:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.app.Activity.getPackageName()' on a null object reference
    at com.microsoft.appcenter.distribute.Distribute.goToUnknownAppsSettings(Distribute.java:1653)
    at com.microsoft.appcenter.distribute.Distribute.access$1000(Distribute.java:112)
    at com.microsoft.appcenter.distribute.Distribute$11.onClick(Distribute.java:1601)
    at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:172)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:193)
    at android.app.ActivityThread.main(ActivityThread.java:6740)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

Maybe a try/catch block would prevent this.

### **Repro Steps**

Unclear.
But likely to happen when the user wants to go to the settings to allow installs from Unknown sources.

### **Details**

1. Which SDK version are you using?
    - 3.3.0
2. Which OS version did you experience the issue on?
    - e.g. Android 9
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Amazon Fire Tablet
4. What third party libraries are you using?
    - N/A
5. Please enable verbose logging for your app using `AppCenter.setLogLevel(Log.VERBOSE)` before your call to `AppCenter.start(...)` and include the logs here:
	- N/A

**Comment by Jamminroot** [comment]
Hey, @fabiendem, thanks for reaching out.

Yeah, it looks that dialog is attempted to be shown while app is in the background.

Please let us some time to investigate. and. possibly, reproduce the problem.

**Comment by fabiendem** [comment]
Thank you 👍

**Comment by vvechkanov** [comment]
Hi! 
Just want to inform you that based on the code we found a potential problem and working on a fix.

**Comment by fabiendem** [comment]
Thank you, yeah am watching https://github.com/microsoft/appcenter-sdk-android/pull/1479 😄

**Comment by AnastasiaKubova** [comment]
The fix has been released in version `4.1.0`. Please update the App Center SDK to the [latest version](https://github.com/microsoft/appcenter-sdk-android/releases/tag/4.1.0) and let me know if this issue still persists.

**Comment by build3r** [comment]
This issue still persist on v 4.1.0
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources$Theme android.content.Context.getTheme()' on a null object reference
       at android.app.AlertDialog.resolveDialogTheme(AlertDialog.java:247)
       at android.app.AlertDialog$Builder.<init>(AlertDialog.java:477)
       at com.microsoft.appcenter.distribute.Distribute.showUnknownSourcesDialog(Distribute.java:1574)
       at com.microsoft.appcenter.distribute.Distribute.enqueueDownloadOrShowUnknownSourcesDialog(Distribute.java:1717)
       at com.microsoft.appcenter.distribute.Distribute$6.onClick(Distribute.java:1480)
       at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:182)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:200)
       at android.app.ActivityThread.main(ActivityThread.java:6971)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
 ```
 using version 
 ```
  implementation "com.microsoft.appcenter:appcenter-distribute:4.1.0"
  ```

**Comment by AnastasiaKubova** [comment]
Hi @build3r ! Thanks for getting in touch with us. Could you please share the reproduction steps?

**Comment by build3r** [comment]
This error happened on beta testers device and captured on crashlytics, sorry I'm not able to reproduce it locally.
I'm using the v 4.1.0 which the latest as per release page.
Dependency screenshot:
<img width="437" alt="image" src="https://user-images.githubusercontent.com/6481548/108701615-326ece00-752e-11eb-871b-b6429a4fa934.png">

Also today saw new error from apcenter SDK happened on a Xiaomi device
```
Caused by java.lang.IllegalArgumentException: Unknown URL content://downloads/my_downloads
       at android.content.ContentResolver.insert(ContentResolver.java:1599)
       at android.app.DownloadManager.enqueue(DownloadManager.java:1163)
       at com.microsoft.appcenter.distribute.download.manager.DownloadManagerRequestTask.doInBackground(DownloadManagerRequestTask.java:45)
       at com.microsoft.appcenter.distribute.download.manager.DownloadManagerRequestTask.doInBackground(DownloadManagerRequestTask.java:21)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
  ```

**Comment by build3r** [comment]
I think the check
`if (mForegroundActivity == null)` 
happening on line [L1661](https://github.com/microsoft/appcenter-sdk-android/blob/d01d7fec1b41e4280ccec54806543cee6f7587d6/sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java#L1661) should happen before we even create the dialog and show it.

**Comment by AnastasiaKubova** [comment]
> Caused by java.lang.IllegalArgumentException: Unknown URL content://downloads/my_downloads

Could you please clarify what the android version use here? Does this issue perform only on Xiaomi devices or this issue happens on other devices also?

**Comment by build3r** [comment]
@AnastasiaKubova 
this error
> Caused by java.lang.IllegalArgumentException: Unknown URL content://downloads/my_downloads

seems like its occuring only in Xiaomi. Issue observed in Redmi 7, Redmi 6 Pro
The OS version is Android 9
![image](https://user-images.githubusercontent.com/6481548/108989423-9377dc80-76bb-11eb-85d4-fc4654009522.png)

However the Distribute dialog error is happening on multiple devices
one of the most recent crash, the line number matches in this stacktrace
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources$Theme android.content.Context.getTheme()' on a null object reference
       at android.app.AlertDialog.resolveDialogTheme(AlertDialog.java:227)
       at android.app.AlertDialog$Builder.<init>(AlertDialog.java:478)
       at com.microsoft.appcenter.distribute.Distribute.showUnknownSourcesDialog(Distribute.java:1584)
       at com.microsoft.appcenter.distribute.Distribute.enqueueDownloadOrShowUnknownSourcesDialog(Distribute.java:1731)
       at com.microsoft.appcenter.distribute.Distribute$6.onClick(Distribute.java:1490)
       at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:191)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:7948)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1075)
```
       
![image](https://user-images.githubusercontent.com/6481548/108989882-1bf67d00-76bc-11eb-80d1-4e1553361c10.png)

**Comment by AnastasiaKubova** [comment]
Thanks for the details, I reopen this ticket.

**Comment by AnastasiaKubova** [comment]
Hi there! The fixes have been released in the [4.1.1](https://github.com/microsoft/appcenter-sdk-android/releases/tag/4.1.1) version of App Center SDK. Please pay attention that due to the termination of `jCenter` support all our assemblies were moved to the `Maven Central` repository.

See more about migration to `Maven Central` [here](https://docs.microsoft.com/en-us/appcenter/sdk/getting-started/android#3-add-the-app-center-sdk-modules).
